electrum/lnpeer.py

```diff
@@ -2734,14 +2734,19 @@ async def wrapped_callback():
                     fut = asyncio.ensure_future(wrapped_callback())
                 # return payment_key so this branch will not be executed again
                 return None, payment_key, None
+            elif preimage:
+                return preimage, None, None
+            else:
+                # we are waiting for mpp consolidation or preimage
+                return None, None, None
         else:
-            assert payment_key == forwarding_key
             # HTLC we are supposed to forward, and have already forwarded
+            # for trampoline onions, forwarding failures are stored with forwarding_key (which is the inner key)
```

Comment:
what exactly do you mean here? If we are "trampoline-forwarding", then we don't have access to the inner key, I think.
Do you mean "for trampoline onion for which we were the final recipient but are still doing stuff that we call *forwarding*" (e.g. a swap)?

Reply:
yes, I meant the latter